rust/foxglove/src/sink.rs

```diff
@@ -10,6 +10,11 @@ use crate::{ChannelId, FoxgloveError, RawChannel};
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord)]
 pub struct SinkId(u64);
```

Comment:
If we change this to NonZeroU64 it will be 64 bits when wrapped with Optional. It also makes it a little clearer that zero is an invalid value.

Cbindgen will be able to covert a u64 to an optional<sinkid>. So we use u64 on the C side and optional on the rust side.